### PR TITLE
Simplify UdpSocket futures

### PR DESCRIPTION
**Pull request by cramertj**
This is just some cleanup that makes the code easier to read for me personally. Feel free to close this if you think it isn't an improvement. I'd been pointing some people at this code as an example of how to write non-blocking futures with tokio-core, and they found the old code hard to parse.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks! I think I may have accidentally merged in the wrong order creating conflicts though, want to rebase?

**Comment by cramertj** [comment]
Nah, no problem-- I don't think there was any order in which you could have merged them that wouldn't have caused conflicts.
